docs/Design/Rest Parameters.md

```diff
@@ -0,0 +1,125 @@
+# Rest Parameters
+
+A parameter that creates an array for all unassigned arguments.
+
+
+## Commands
+
+### `rest parameters`
+
+`rest parameters : type, name`
+
+Creates a rest parameter.
+The name is the name of the array created for the unassigned arguments.
+
+## Usage
+
+```gls
+rest parameters : string, foo
+```
+
+### CSharp
+
+```csharp
+params string[] foo
+```
+
+### Java
+
+```java
+string... foo
+```
+### Python
+
+```python
+*foo
+```
+
+### Ruby
+
+```ruby
+*foo
+```
+
+### TypeScript
+
+```typescript
+...foo

```

Comment:
The GLS preference for TypeScript is to type everything. This should be `...foo: string[]`